spec/meilisearch/index/search/filters_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+RSpec.describe 'MeiliSearch::Index - Search' do
+  before(:all) do
+    @documents = [
+      { objectId: 123,  title: 'Pride and Prejudice',                    genre: 'romance' },
+      { objectId: 456,  title: 'Le Petit Prince',                        genre: 'adventure' },
+      { objectId: 1,    title: 'Alice In Wonderland',                    genre: 'adventure' },
+      { objectId: 2,    title: 'Le Rouge et le Noir',                    genre: 'romance' },
+      { objectId: 1344, title: 'The Hobbit',                             genre: 'adventure' },
+      { objectId: 4,    title: 'Harry Potter and the Half-Blood Prince', genre: 'fantasy' },
+      { objectId: 42,   title: 'The Hitchhiker\'s Guide to the Galaxy' }
+    ]
+    client = MeiliSearch::Client.new($URL, $MASTER_KEY)
+    clear_all_indexes(client)
+    @index = client.create_index('books')
+    @index.add_documents(@documents)
+    sleep(0.1)
+  end
+
+  after(:all) do
+    @index.delete
+  end
+
+  let(:default_search_response_keys) do
+    [
+      'hits',
+      'offset',
+      'limit',
+      'nbHits',
+      'exhaustiveNbHits',
+      'processingTimeMs',
+      'query'
+    ]
+  end
+
+  it 'does a custom search with filters' do
+    response = @index.search('le', { filters: 'genre = romance' })
+    expect(response['hits'].count).to eq(1)
+    expect(response['hits'].first['objectId']).to eq(2)
+  end
+
+  it 'does a custom placeholder search with filters' do
+    response = @index.search('', { filters: 'genre = adventure' })
+    expect(response['hits'].count).to eq(3)
+  end
```

Comment:
Can you add more fields in `@documents` to make more complex search tests with filters please? (with AND and OR)